### PR TITLE
Filter out from watch list files in `.spago` folder

### PR DESCRIPTION
**Pull request by Benjmhart**
### Description of the change

TODO

### Checklist:

- [X] Added the change to the "Unreleased" section of the changelog
- [ ] Added some example of the new feature to the `README`
- [ ] Added a test for the contribution (if applicable)

**P.S.**: the above checks are not compulsory to get a change merged, so you may skip them. However, taking care of them will result in less work for the maintainers and will be much appreciated 😊

### CODE REVIEW

**Comment by Benjmhart** [comment]
original issue https://github.com/spacchetti/spago/issues/430

File: src/Spago/Build.hs [hunk 1]
```diff
@@ -137,6 +137,7 @@ build buildOpts@BuildOptions{..} maybePostBuild = do
 
     wrap   = Purs.SourcePath . Text.pack
     unwrap = Text.unpack . Purs.unSourcePath
+    removeDotSpago = filter $ not . (".spago" `List.isInfixOf`)
```
Comment (jhrcek) [inline src/Spago/Build.hs]: This approach seems reasonable to me. There's however one edge case that comes to my mind:
For example path like this `/home/John/my.spago.playground/my.project/` will be removed by this filter too. This particular  case could be prevented by treating `.spago` as directory name (see [splitDirectories](https://www.stackage.org/haddock/lts-14.7/filepath-1.4.2.1/System-FilePath-Posix.html#v:splitDirectories)) as opposed to infix of a string (`my.spago.playground`). But maybe I'm just too cautious :laughing: 

```suggestion
    removeDotSpago = filter (\glob -> ".spago" `notElem` splitDirectories glob)
```

Comment (jhrcek) [reply src/Spago/Build.hs]: Sorry, I just did one-line suggestion as that's what github comments UI allows. You'll need to add import to make the CI work.

Comment (Benjmhart) [reply src/Spago/Build.hs]: Thanks for the heads up,   I'm running a local build to verify and then I'll update the PR with code that actually works.

**Pull request merged**